pep-0677.rst

```diff
@@ -635,24 +635,107 @@ callable types and ``=>`` for lambdas.
 Runtime Behavior
 ----------------
 
-Our tentative plan is that:
-
-- The ``__repr__`` will show an arrow syntax literal.
-- We will provide a new API where the runtime data structure can be
-  accessed in the same manner as the AST data structure.
-- We will ensure that we provide an API that is backward-compatible
-  with ``typing.Callable`` and ``typing.Concatenate``, specifically
-  the behavior of ``__args__`` and ``__parameters__``.
+The new AST nodes need to evaluate to runtime types, and we have two goals for the
+behavior of these runtime types:
+
+- They should expose a structured API that is descriptive and powerful
+  enough to be compatible with extending the type to include new features
+  like named and variadic arguments.
+- They should also expose an API that is fully backward-compatible with
```

Comment:
I'd drop "fully". After all this is just a transitional convenience, no code breaks unless users explicitly change `Callable[[X, Y], Z]` into `(X, Y) -> Z`. In general code that picks apart Callables is probably used to type annotation internals changing between releases. Also, I expect that in most cases such code will have to be adjusted to look for the new callable type anyway, few people will write code that just "dives in" and does `f.__annotations__["callback"].__args__`.

```suggestion
- They should also expose an API that is backward-compatible with
```